presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -67,16 +71,25 @@ public HiveMetastoreClient createMetastoreClient()
         List<HostAndPort> metastores = new ArrayList<>(addresses);
         Collections.shuffle(metastores.subList(1, metastores.size()));
 
-        TTransportException lastException = null;
+        TException lastException = null;
         for (HostAndPort metastore : metastores) {
             try {
-                return clientFactory.create(metastore);
+                HiveMetastoreClient hiveMetastoreClient = clientFactory.create(metastore);
+                if (!isNullOrEmpty(hiveUserName)) {
+                    hiveMetastoreClient.setUGI(hiveUserName, new ArrayList<String>());
+                }
+                return hiveMetastoreClient;
             }
-            catch (TTransportException e) {
```

Comment:
If we get a `TException` that's not a `TTransportException` from the `setUGI` call, that is likely to be a permanent error (metastore doesn't allow that user). Do we want to try the next metastore in that case, or abort immediately?

Reply:
This is a good question. This need to be based on Metastore return information to do processing, Metastore set_ugi not dealt with. This is the Metastore implementation
```
public List<String> set_ugi(String username, List<String> groupNames) throws TException {
      Collections.addAll(groupNames, username);
      return groupNames;
}
```
Do you have any good suggestions?